src/main/java/com/amazonaws/secretsmanager/caching/cache/SecretCacheObject.java

```diff
@@ -229,9 +230,12 @@ public boolean refreshNow() throws InterruptedException {
         Thread.sleep(sleep);
```

Comment:
I haven't had the opportunity to explore how Virtual Threads work, does the regular Thread class `sleep()` call still work the same way with them?

Reply:
The regualar Thread.sleep() still works here. I tested it and didn't find problem.  
The problem only occurs when the synchronized block has blocking operation.  
You can reproduce the issue in Linux environment with the code in https://github.com/aws/aws-secretsmanager-caching-java/issues/115.